hass_nabucasa/__init__.py

```diff
@@ -169,13 +183,23 @@ async def update_token(
         id_token: str,
         access_token: str,
         refresh_token: str | None = None,
+        *,
+        check_connection: bool = False,
     ) -> asyncio.Task | None:
         """Update the id and access token."""
         self.id_token = id_token
         self.access_token = access_token
         if refresh_token is not None:
             self.refresh_token = refresh_token
 
+        if (
+            check_connection
+            and (connection := await self._get_connection(skip_token_check=True))[
+                "connected"
+            ]
+        ):
+            raise AlreadyConnectedError(details=connection["details"])
```

Comment:
Why it's this here instead of at the beginning of the `login` method? 
I see that we make all the checks in here, but seems a mix of responsibilities to perform them under the `update_token` method..

Maybe we should move them to a separate method?

Reply:
It is here because we need to partially set the attributes which happens here.